src/workers/newNotificationV2Mail.ts

```diff
@@ -132,9 +132,65 @@ export type TemplateDataFunc = (
   avatars: NotificationAvatarV2[],
 ) => Promise<TemplateData | null>;
 const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
-  post_boost_completed: generateBoostEmailUpdate,
-  post_boost_first_milestone: generateBoostEmailUpdate,
-  campaign_completed: generateCampaignCompletedEmail,
+  campaign_post_completed: async (con, user, notification) => {
+    const campaign = await con
+      .getRepository(CampaignPost)
+      .findOneBy({ id: notification.referenceId });
+
+    if (!campaign) {
+      return null;
+    }
+
+    const post = await con.getRepository(Post).findOneOrFail({
+      where: { id: campaign.postId },
+    });
```

Comment:
potentially preload post on campaign query above with relation

Reply:
Right. Let me see if we can also preload the sharedPost 🤯